### PR TITLE
Byte typed addresses

### PR DESCRIPTION
**Pull request by karlb**
This will need some cleanups, but those should be easier to do now that the main change is done. And I don't want to wait with the merge since this will quickly attract merge conflicts during rebases.

Closes https://github.com/raiden-network/raiden-services/issues/325

### CODE REVIEW

File: src/monitoring_service/database.py [hunk 1]
```diff
@@ -102,23 +106,29 @@ def upsert_channel(self, channel: Channel) -> None:
...
-            [hex(channel_id), token_network_address],
+            [hex(channel_id), to_checksum_address(token_network_address)],
         ).fetchone()
 
         if row is None:
             return None
         kwargs = {
             key: val for key, val in zip(row.keys(), row) if not key.startswith("update_status")
         }
+        kwargs["token_network_address"] = decode_hex(kwargs["token_network_address"])
```
Comment (palango) [inline src/monitoring_service/database.py]: Mh, I'd perfer to use `to_canonical_address` for hex to bytes conversion. I think it makes the intention clearer.

File: src/pathfinding_service/api.py [hunk 2]
```diff
@@ -42,11 +41,15 @@
 
 class ApiWithErrorHandler(Api):
     def handle_error(self, e: exceptions.ApiException) -> Response:
-        log.debug("Error while handling request", error=e, details=e.error_details, message=e.msg)
-        return self.make_response(
-            {"errors": e.msg, "error_code": e.error_code, "error_details": e.error_details},
-            e.http_code,
-        )
+        if isinstance(e, exceptions.ApiException):
```
Comment (palango) [inline src/pathfinding_service/api.py]: Why do we need this check? Following the annotation this should always be true.

File: src/pathfinding_service/api.py [hunk 3]
```diff
@@ -263,24 +266,23 @@ def get(self) -> Tuple[dict, int]:
...
         )
 
 
 class DebugEndpoint(PathfinderResource):
     def get(  # pylint: disable=no-self-use
-        self,
-        token_network_address: TokenNetworkAddress,
-        source_address: Address,
-        target_address: Address = None,
+        self, token_network_address: str, source_address: Address, target_address: Address = None
```
Comment (palango) [inline src/pathfinding_service/api.py]: `source_address` and `target_address` should be strings as well.

File: src/pathfinding_service/api.py (hunk 2 above)
Comment (karlb) [reply src/pathfinding_service/api.py]: The annotation should be more generic. We might get any kind of exception in the error handler, AFAICS.

**Comment by palango** [comment]
Another point, should we switch the DB representation of addresses to bytes as well? It would make stuff more consistent, but would be harder to query by hand.

File: src/monitoring_service/database.py (hunk 1 above)
Comment (karlb) [reply src/monitoring_service/database.py]: But that returns eth_typing.Address, which we are not using, so I expect that we'll get type errors. Do you agree?

And why don't we use eth_typing.Address as address type?

**Comment by karlb** [comment]
> Another point, should we switch the DB representation of addresses to bytes as well? It would make stuff more consistent, but would be harder to query by hand.

I thought about this, too. It would work fine from an sqlite point of view, but I'm still torn due to the readbility and also writeability when manually dealing with the db.

File: src/monitoring_service/database.py (hunk 1 above)
Comment (palango) [reply src/monitoring_service/database.py]: In a quick check mypy didn't complain, which is a bit surprising.

And I think the `eth_typing` package is quite new, so we might want to use the Raiden types as aliases.

Comment (palango) [reply src/monitoring_service/database.py]: Anyways, we can fix the `decode_hex` stuff later.

**Comment by palango** [comment]
> readbility and also writeability when manually dealing with the db.

Yeah, let's keep it readable for now. The DB code shouldn't change that much anyways.

**Pull request merged**